src/superphot_plus/ztf_transient_fit.py

```diff
@@ -83,45 +83,55 @@ def run_mcmc(lc, t0_lim=None, plot=False, rstate=None):
         Flag to enable/disable plotting. Defaults to False.
     rstate : int, optional
         Random state that is seeded. if none, use machine entropy.
+    telescope : str, optional
+        Determines band and prior information. Defaults to ZTF.
 
     Returns
     -------
     np.ndarray or None
         Numpy array containing the equally weighted posteriors, or None
         if the data is invalid.
     """
-    ref_band_idx = 1  # red band # pylint: disable=unused-variable
-    n_params = 14
+    if telescope == "ZTF":
+        all_priors_cls = MultibandPriors.load_ztf_priors()
+        all_priors = all_priors_cls.to_numpy().T
+        ref_band = "r" # maybe define within MultibandPriors class
+
+    n_params = len(all_priors.T)
+    unique_bands = all_priors_cls.ordered_bands
+    ref_band_idx = np.argmax(unique_bands == ref_band)
 
     # Require data in both the g and r bands.
-    if lc.obs_count("r") == 0 or lc.obs_count("g") == 0:
-        return None
+    for ub in unique_bands:
+        if lc.obs_count(ub) == 0:
+            return None
 
     tdata = lc.times
     fdata = lc.fluxes
     ferrdata = lc.flux_errors
     bdata = lc.bands
 
     # Precompute the information about the maximum flux in the r band.
-    r_band = bdata == "r"
-    max_index = np.argmax(lc.fluxes[r_band] - np.abs(lc.flux_errors[r_band]))
-    max_flux = lc.fluxes[r_band][max_index] - np.abs(lc.flux_errors[r_band][max_index])
-    max_flux_loc = tdata[r_band][max_index]
+    where_ref_band = (bdata == ref_band)
+    max_index = np.argmax(lc.fluxes[where_ref_band] - np.abs(lc.flux_errors[where_ref_band]))
```

Comment:
Not for this PR: Is this something that would be worth pulling into Lightcurve (something like `def max_flux_index(self, band=...)` )? I considered doing that when creating Lightcurve, but it didn't seem like it was needed in too many places. But if it will be used in other places, it would be a useful thing to add in a future PR.

Reply:
Yes, I also was thinking about this too! I will definitely add as an Issue that anyone can work on